reactive/jenkins.py

```diff
@@ -173,6 +180,49 @@ def migrate_charm_data():
     configuration.migrate()
 
 
+@when('nrpe-external-master.available')
+def update_nrpe_config(nagios):
+    udata = unitdata.kv()
+    nagios_hostname = udata.get('nagios.hostname', None)
+    nagios_host_context = udata.get('nagios.host_context', None)
+
+    # require the nrpe-external-master relation to provide the host context
+    if in_relation_hook() and relation_id().\
+            startswith('nrpe-external-master:'):
+        rel = relation_get()
+        if 'nagios_host_context' in rel:
+            nagios_host_context = rel['nagios_host_context']
+            udata.set('nagios.host_context', nagios_host_context)
+
+            # We have to strip the nagios host context from the nagios hostname
+            # since the nagios.add_check will put it back again...
+            nagios_hostname = rel['nagios_hostname']
+            if nagios_hostname.startswith(nagios_host_context + '-'):
+                nagios_hostname = nagios_hostname[len(nagios_host_context +
+                                                      '-'):]
+
+            udata.set('nagios.hostname', nagios_hostname)
+
+    if not nagios_hostname or not nagios_host_context:
+        return
```

Comment:
In this case, please open an issue against https://github.com/cmars/nrpe-external-master-interface and include the narrative that you just exposed. Including the boilerplate code for reference would be good to.

Then, please add a comment at the beginning of the hook mentioning that the boilerplate should eventually go away, e.g.:

```python
# XXX This boilerplate is needed until:
#
# https://github.com/cmars/nrpe-external-master-interface/issues/N
# gets fixed.
```